### PR TITLE
Disable link to unavailable pacemaker cluster detail in home/health summary

### PR DESCRIPTION
**Pull request by nelsonkopliku**
# Description

This PR disables links to pacemaker cluster in the dashboard when a pacemaker cluster is not available.

no-cluster-link

## How was this tested?

An additional jest test was added and it was tested manually against the scenario provided by Alberto.

### CODE REVIEW

**Comment by CDimonaco** [comment]
I think we have the "core" functionality right here, but the hovering animation is still present, I think we should disable the hover transition and the pointer events, so that the item can completely act as a "non button", what do you think? Tag @jagabomb for further UI insights 👁️

File: assets/js/components/HealthSummary/HomeHealthSummary.jsx [hunk 1]
```diff
@@ -52,7 +52,10 @@ const healthSummaryTableConfig = {
       render: (content, item) => {
         const linkToCluster = `/clusters/${item.clusterId}`;
```
Comment (EMaksy) [inline assets/js/components/HealthSummary/HomeHealthSummary.jsx:53]: Hey can we use an if statement to check if there is an clusterId? 
If there is none, we don't render the Link. 
Otherwise, we just hide the rendered Link.

What do you think ?

**Review by EMaksy — CHANGES_REQUESTED** [review]

**Comment by nelsonkopliku** [comment]
> I think we have the "core" functionality right here, but the hovering animation is still present, I think we should disable the hover transition and the pointer events, so that the item can completely act as a "non button"

hey @CDimonaco I agree with you, the fact is that to me it seems to be already behaving that way :smile: 

The hovering animation you are referring to is bound to `HealthIcon` component, and by disabling `pointerEvents` on the outer `Link`, those are not propagated to the contained icon and there is no hovering.

This is at least what I can observe when running it locally and in the attached gif

What am I missing?

File: assets/js/components/HealthSummary/HomeHealthSummary.jsx (hunk 1 above)
Comment (nelsonkopliku) [reply assets/js/components/HealthSummary/HomeHealthSummary.jsx:53]: As discussed, that would mean having something like
```javascript
if (!item?.clusterId) {
  # here the HealthIcon needs to be instructed about the hover opacity
  return <HealthIcon health={content} centered={true} hoverOpacity={'0'}/>;
}

const linkToCluster = `/clusters/${item.clusterId}`;
return (
  <Link to={linkToCluster}>
    <HealthIcon health={content} centered={true} />
  </Link>
);
```
which I'd be fine with, yet it would require making an extra change to the `HealthIcon` component to accept the opacity, because it defaults to `hover:opacity-75` while in a specific scenario has to be 0.

I find enough disabling `pointerEvents` when we do not have a clusterId, keeping the scope narrow.
Still open to other's stance on this.

Comment (dottorblaster) [reply assets/js/components/HealthSummary/HomeHealthSummary.jsx:53]: > I find enough disabling pointerEvents when we do not have a clusterId

We don't change how event propagation works for a corner case. We change the rendered DOM instead. That's the point about React.

@EMaksy is right, we'll have to modify the `HealthIcon` using a prop accordingly, maybe a boolean one. Also, you might find useful what React docs say about conditional rendering :smile: 

https://reactjs.org/docs/conditional-rendering.html

Comment (nelsonkopliku) [reply assets/js/components/HealthSummary/HomeHealthSummary.jsx:53]: right, already two in that direction. Changing it!

File: assets/js/components/HealthSummary/HomeHealthSummary.test.jsx [hunk 2]
```diff
@@ -53,6 +60,44 @@ describe('HomeHealthSummary component', () => {
...
+      container
+        .querySelector(unavailablePacemakerClusterSelector)
+        .getAttribute('href')
+    ).toContain(`/clusters/null`);
+
+    expect(
+      container.querySelector(unavailablePacemakerClusterSelector).style
+        .pointerEvents
+    ).toBe('none');
+  });
```
Comment (dottorblaster) [inline assets/js/components/HealthSummary/HomeHealthSummary.test.jsx:100]: Actually I would expect no href to be present if the cluster id is null :sweat_smile:

File: assets/js/components/HealthSummary/HomeHealthSummary.jsx [hunk 3]
```diff
@@ -52,7 +52,10 @@ const healthSummaryTableConfig = {
...
         return (
-          <Link to={linkToCluster}>
+          <Link
+            to={linkToCluster}
+            style={{ pointerEvents: item?.clusterId ? 'auto' : 'none' }}
+          >
             <HealthIcon health={content} centered={true} />
           </Link>
```
Comment (dottorblaster) [inline assets/js/components/HealthSummary/HomeHealthSummary.jsx:60]: You can use a ternary and render the link conditionally

```suggestion
          item.clusterId ? (
            <Link
              to={linkToCluster}
              style={{ pointerEvents: item?.clusterId ? 'auto' : 'none' }}
            >
              <HealthIcon health={content} centered={true} />
            </Link>
          ) : (
            <HealthIcon health={content} centered={true} />
          )
```

**Review by dottorblaster — CHANGES_REQUESTED** [review]

File: assets/js/components/HealthSummary/HomeHealthSummary.jsx (hunk 1 above)
Comment (nelsonkopliku) [reply assets/js/components/HealthSummary/HomeHealthSummary.jsx:53]: Actually let me rephrase: changing because you are right. 

It semantically does not make sense having a Link which is actually not going to be a Link.
I overlooked that semantics, my bad :sweat_smile: 

Thanks guys! :heart:

File: assets/js/components/Health/HealthIcon.jsx [hunk 4]
```diff
@@ -11,13 +11,22 @@ import {
...
-const HealthIcon = ({ health = undefined, centered = false }) => {
+const HealthIcon = ({
+  health = undefined,
+  centered = false,
+  hoverOpacity = true,
+}) => {
+  const hoverOpacityClass = {
+    'hover:opacity-75': hoverOpacity,
+    'hover:opacity-100': !hoverOpacity,
+  };
```
Comment (dottorblaster) [inline assets/js/components/Health/HealthIcon.jsx:22]: Your call but keep in mind that you can also just pass `className` down to the rendered component

**Review by dottorblaster — APPROVED** [review]
LGTM

**Review by EMaksy — APPROVED** [review]
LGTM, thanks for applying the changes :+1:
